### PR TITLE
Some minor improvements suggested from Ned's use

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Adjusted `AWS_Credential` to have a `Default` and removed support for `Nothing` from functions.
- Renamed `Response_Body.to_file` to `Response_Body.write`.
- Add `write` to `Response`.
- Add `Table.get_value` and `DB_Table.get_value` allowing getting a single value from a table.
- Added `Data.download` allowing downloading from a URL to a file.
- Added text widget as input widget for `Data` methods.

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso [hunk 1]
```diff
@@ -30,12 +30,12 @@ type S3_File
          If the path contains `.` or `..` segments, they will be normalized.
        - credentials: The credentials to use when accessing the file.
          If not specified, the default credentials are used.
-    new : Text -> AWS_Credential | Nothing -> S3_File ! Illegal_Argument
-    new (uri : Text = S3.uri_prefix) (credentials : AWS_Credential | Nothing = Nothing) =
+    new : Text -> AWS_Credential -> S3_File ! Illegal_Argument
+    new (uri : Text = S3.uri_prefix) credentials:AWS_Credential=AWS_Credential.Default =
```
Comment (radeusgd) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso:34]:
```suggestion
    new (uri : Text = S3.uri_prefix) (credentials : AWS_Credential = AWS_Credential.Default) =
```

nit: IMHO the version with spaces + parentheses is more readable

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/DB_Table.enso [hunk 2]
```diff
@@ -146,6 +146,21 @@ type DB_Table
...
+       ICON select_column
+       Gets a value from the table.
+
+       Arguments:
+       - selector: The name or index of the column.
+       - index: The index of the value to get within the column.
+       - if_missing: The value to use if the selector isn't present.
+    @selector Widget_Helpers.make_column_name_selector
+    @index (t-> Widget.Numeric_Input minimum=0 maximum=t.row_count-1)
+    get_value : Text | Integer -> Integer -> Any -> Any
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/DB_Table.enso:159]: Why was `get_value` chosen over `get_cell`?

**Review by radeusgd — APPROVED** [review]
Looks great

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/DB_Table.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Data/DB_Table.enso:159]: bounced back and forth with Ned and we felt value was better but it was pretty equal.

File: distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso:34]: will do as part of the ICON apply this evening.

**Pull request merged**